danger/Dangerfile

```diff
@@ -0,0 +1,27 @@
+# Check if branch name follows format
+type_pattern = /(fix|feat|refactor|improve|build|ci|docs|chore|test|tests|revert)/
+ticket_pattern = /(MAG|SDKCF)-\d{3,5}/ # Main single ticket in capital letters that can connect this branch to a board tracker ID
+desc_pattern = /[a-z0-9]+(?:-+[a-z0-9]+)*$/ # Short description in small letters and separated by dashes to easily identify the purpose of branch at a glance
+branch_name_pattern_1 = /^#{type_pattern}\/#{ticket_pattern}_#{desc_pattern}/
+branch_name_pattern_2 =  /^#{type_pattern}\/#{desc_pattern}/
+branch_name_pattern_3 = /^release\/[\w]+/ # Less stricter check for release branches (can contain special releases not only using version name)
+
+branch_name = github.branch_for_head
```

Comment:
>bundle exec danger dry_run --verbose
>[!] Invalid `Dangerfile` file: undefined local variable or method `github’ for #

If you do a Danger local/dry run, this will fail because the `github` var is only exposed when you do a PR against GitHub. In our iOS Dangerfile, we got around this by adding a `if defined?(github)` condition. This may or may not be an issue for your team. Just FYI

Reply:
Thanks for the info. I think none of the members' using `dry_run` as of now. We only test Dangerfile changes using `pr`. We'll leave it as is for now. Thanks